File: src/semana-1/Count-Items-Matching-a-Rule.cpp
#include <bits/stdc++.h>
using namespace std;
// https://leetcode.com/problems/count-items-matching-a-rule/

class Solution
{
public:
    int countMatches(vector<vector<string>> &items, string ruleKey, string ruleValue)
    {
        int rk, count = 0;
        if (ruleKey == "type")
            rk = 0;
        if (ruleKey == "color")
            rk = 1;
        if (ruleKey == "name")
            rk = 2;
        
        for (auto i : items)
        {
            if(i.at(rk) == ruleValue)
                count++;
        }
        return count;

    }
};

int main()
{
    vector<vector<string>> test;
    test.resize(2);
    test.at(0).push_back("olá mundo");
    test.at(0).push_back("mundo olá");
    cout << test.at(0).at(0) << endl;
    cout << test.at(0).at(1) << endl;
    return 0;
}

File: src/semana-1/JavaScript/Count-Items-matching-a-Rule.js
/**
 * @param {string[][]} items
 * @param {string} ruleKey
 * @param {string} ruleValue
 * @return {number}
 * https://leetcode.com/problems/count-items-matching-a-rule/
 */
 const items  = [["phone","blue","pixel"],["computer","silver","phone"],["phone","gold","iphone"]]; 
 const ruleKey = "type"; 
 const ruleValue = "phone";

var countMatches = function(items, ruleKey, ruleValue) {
  var rk = 0;

  switch (ruleKey) {
    case "type":
      rk = 0;
      break;

    case "color":
      rk = 1;
      break;

    case "name":
      rk = 2;
      break;
  
    default:
      break;
  }
  
  return items.filter(item => item[rk] === ruleValue).length;
};

console.log(countMatches(items, ruleKey, ruleValue));